invenio_app_rdm/theme/assets/semantic-ui/js/invenio_app_rdm/administration/records/CompareRevisions.js

```diff
@@ -0,0 +1,165 @@
+/*
+ * // This file is part of Invenio-App-Rdm
+ * // Copyright (C) 2025 CERN.
+ * //
+ * // Invenio-App-Rdm is free software; you can redistribute it and/or modify it
+ * // under the terms of the MIT License; see LICENSE file for more details.
+ */
+
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import { RecordModerationApi } from "./api";
+import { withCancel, ErrorMessage } from "react-invenio-forms";
+import { Modal, Grid, Segment, Button } from "semantic-ui-react";
+import { i18next } from "@translations/invenio_app_rdm/i18next";
+import { Differ, Viewer } from "json-diff-kit";
+import { CompareRevisionsDropdown } from "../components/CompareRevisionsDropdown";
+
+export class CompareRevisions extends Component {
+  constructor(props) {
+    super(props);
+    this.differ = new Differ({
+      detectCircular: true,
+      maxDepth: null,
+      showModifications: true,
+      arrayDiffMethod: "lcs",
+      ignoreCase: false,
+      ignoreCaseForKey: false,
+      recursiveEqual: true,
+    });
+    this.viewerProps = {
+      indent: 4,
+      lineNumbers: true,
+      highlightInlineDiff: true,
+      inlineDiffOptions: {
+        mode: "word",
+        wordSeparator: " ",
+      },
+      hideUnchangedLines: true,
+      syntaxHighlight: false,
+      virtual: true,
+    };
+
+    this.state = {
+      loading: true,
+      error: undefined,
+      currentDiff: undefined,
+      allRevisions: {},
+      srcRevision: undefined,
+      targetRevision: undefined,
+    };
+  }
+
+  async fetchRevisions() {
+    const { resource } = this.props;
+    this.setState({ loading: true });
+
+    try {
+      this.cancellableAction = withCancel(RecordModerationApi.getRevisions(resource));
+      const response = await this.cancellableAction.promise;
+      const revisions = await response.data;
+
+      this.setState({
+        allRevisions: revisions,
+        targetRevision: revisions[0],
+        srcRevision: revisions.length > 1 ? revisions[1] : revisions[0],
+        loading: false,
+      });
+    } catch (error) {
+      if (error === "UNMOUNTED") return;
+      this.setState({ error: error, loading: false });
+      console.error(error);
+    }
+  }
+
+  componentDidMount() {
+    this.fetchRevisions();
+  }
+
+  componentWillUnmount() {
+    this.cancellableAction && this.cancellableAction.cancel();
+  }
+
+  computeDiff = () => {
```

Comment:
this is where I would separate concerns (what I mentioned in a call earlier), fetching data and computing diff in separate components